cmake/modules/platform/os/zos.cmake

```diff
@@ -20,12 +20,15 @@
 #############################################################################
 
 list(APPEND OMR_PLATFORM_DEFINITIONS
-	-DJ9ZOS390
-	-DLONGLONG
 	-D_ALL_SOURCE
-	-D_XOPEN_SOURCE_EXTENDED
-	-DIBM_ATOE
+	-D_OPEN_THREADS=2
```

Comment:
Why should `_OPEN_THREADS` be `2`? [1] suggests it only needs to be defined.

[1] https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.bpxbd00/ptonce.htm

Reply:
https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.bpxbd00/ftms.htm

>When defined to 1, this indicates that symbols required by POSIX.1, POSIX.1a, and POSIX.4a(draft 6) are made visible.
>
>When defined to 2, additional pthread functions introduced in z/OS V1R07 from Single UNIX Specification, Version 3 are made visible, along with those made visible when this is defined to 1.

Reply:
Elsewhere on that page it uses the word 'tolerate':
> _ALL_SOURCE does not expose functionality first introduced in Single UNIX Specification, Version 3 under macro definitions _XOPEN_SOURCE 600 or _POSIX_C_SOURCE 200112L, although it does tolerate interfaces made visible by defining _OPEN_THREADS to 2 or 3.

This doesn't sound ideal, but I suppose it's necessary because we use `pthread_sigmask()` (and probably other functions) in the list made available when `_OPEN_THREADS == 2`.

Reply:
Is the consensus here that `_OPEN_THREADS == 2` is correct? Can this discussion be resolved?